# Yaml file containing the task istructions for the cooperative manipulation task

# initial robot configuration
q_robot1_home: [0.0, 0.2, 0.0, -1.16, 0.0, 1.27, 1.57, 0.0] # iiwa 7 joints + pivoting
q_robot2_home: [0.0, 0.0, 0.0, 0.26, 0.0, -1.0, 0.0, 0.0] # yaskawa 7 joints + pivoting

# camera transform wrt robot end effector PRE PIVOTING JOINT! 
  #  -1.0000         0         0    0.0240
  #        0         0   1.0000    -0.2590
  #        0    1.0000         0   -0.1559
  #        0         0         0    1.0000 prepivotTcamera iiwa

  #        0   -0.9290   -0.3700    0.1030
  #        0   -0.3700    0.9290   -0.2070
  #  -1.0000         0         0   -0.0020
  #        0         0         0    1.0000 prepivot2Tcamera yaskawa

prepivot1Tcamera: 
  translation: [0.024, -0.2590, -0.1559] 
  quaternion: [0.0, 0.7071,0.7071, 0.0] # [qx, qy, qz, qw]

prepivot2Tcamera:
  translation: [0.103,-0.2070, -0.002] 
  quaternion: [-0.5852,0.3969,0.5852,0.3969] # [qx, qy, qz, qw]


# objects to be manipulated

objects:
  - object:
      name: "resin_block_1"
      bTo_final: # object desired final pose
        translation: [0.8, 0.0, 0.5] 
        quaternion: [ 0.5, -0.5, 0.5, 0.5 ] #[qx, qy, qz, qw] 
      qdes_robot1: [1.57, 0.2, 0.0, -1.16, 0.0, 1.27, 0.0, 0.0] # iiwa 7 joints + pivoting
      qdes_robot2: [0.0, 0.0, 0.0, 0.26, 0.0, -1.0, 0.0, 0.0] # yaskawa 7 joints + pivoting
  - object:
      name: "resin_block_2"
      bTo_final: 
        translation: [0.8, 0.0, 0.5] 
        quaternion: [0.0, -0.7, 0.7, 0.0] # [qx, qy, qz, qw] 
      qdes_robot1: [-1.57, 0.2, 0.0, -1.16, 0.0, 1.27, 0.0, 0.0] # iiwa 7 joints + pivoting
      qdes_robot2: [1.57, 0.0, 0.0, 0.26, 0.0, -1.0, 0.0, 0.0] # yaskawa 7 joints + pivoting